### PR TITLE
[SPARK-15246][Core] Fix code style and improve volatile for SPARK-4452

### PR DESCRIPTION
**Pull request by lianhuiwang**
## What changes were proposed in this pull request?
1. Fix code style  
2. remove volatile of elementsRead method because there is only one thread to use it.
3. avoid volatile of _elementsRead because Collection increases number of  _elementsRead when it insert a element. It is very expensive. So we can avoid it.

After this PR, I will push another PR for branch 1.6.
## How was this patch tested?

unit tests

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/util/collection/Spillable.scala [hunk 1]
```diff
@@ -41,7 +41,7 @@ private[spark] abstract class Spillable[C](taskMemoryManager: TaskMemoryManager)
   protected def forceSpill(): Boolean
 
   // Number of elements read from input since last spill
-  @volatile protected def elementsRead: Long = _elementsRead
+  protected def elementsRead: Long = _elementsRead
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/util/collection/Spillable.scala]: can you comment on why the changes are safe/correct?

**Comment by lianhuiwang** [comment]
cc @davies

File: core/src/main/scala/org/apache/spark/util/collection/Spillable.scala (hunk 1 above)
Comment (lianhuiwang) [inline core/src/main/scala/org/apache/spark/util/collection/Spillable.scala]: It has only one thread to use this method.  So we can remove volatile.

File: core/src/main/scala/org/apache/spark/util/collection/Spillable.scala [hunk 2]
```diff
@@ -112,7 +112,6 @@ private[spark] abstract class Spillable[C](taskMemoryManager: TaskMemoryManager)
       if (!isSpilled) {
         0L
       } else {
-        _elementsRead = 0
```
Comment (lianhuiwang) [inline core/src/main/scala/org/apache/spark/util/collection/Spillable.scala]: I delete this code in order to _elementsRead is called by one thread. So we do not make _elementsRead  volatile.

**Comment by davies** [comment]
Is this really an hot fix (serious bug or breaking tests)?

**Comment by lianhuiwang** [comment]
@davies I think it is not an hot fix. So i have updated PR. thanks.

**Comment by davies** [comment]
LGTM

File: core/src/main/scala/org/apache/spark/util/collection/Spillable.scala [hunk 3]
```diff
@@ -41,7 +41,7 @@ private[spark] abstract class Spillable[C](taskMemoryManager: TaskMemoryManager)
   protected def forceSpill(): Boolean
 
   // Number of elements read from input since last spill
-  @volatile protected def elementsRead: Long = _elementsRead
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/util/collection/Spillable.scala]: It's not meaningful to make a method volatile anyway right? the fact that it's not an error makes me think I could be missing something. No objection

Comment (lianhuiwang) [inline core/src/main/scala/org/apache/spark/util/collection/Spillable.scala]: Yes, I think you understand correctly.

**Comment by rxin** [comment]
Merging in master/2.0. Thanks.
